### PR TITLE
[Ch11957] Change exception with errorcode 500 to ValidationError on duplicate SSFA add

### PR DESCRIPTION
**Pull request by csabadenes**
[ch11957]

### CODE REVIEW

File: src/etools/applications/partners/views/interventions_v2.py [hunk 1]
```diff
@@ -146,6 +147,15 @@ def create(self, request, *args, **kwargs):
             'result_links'
         ]
         nested_related_names = ['ll_results']
+
+        if request.data.get('document_type') == Intervention.SSFA:
+            agreement = Agreement.objects.get(pk=request.data.get('agreement'))
+            if agreement and agreement.interventions.all().count():
```
Comment (domdinicola) [inline src/etools/applications/partners/views/interventions_v2.py]: pls remove .all()

File: src/etools/applications/partners/tests/test_views.py [hunk 2]
```diff
@@ -1420,6 +1420,31 @@ def test_intervention_validation_doctype_ssfa(self):
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
         self.assertIn('Document type PD or HPD can only be associated with a PCA agreement.', response.data)
 
+    def test_intervention_validation_multiple_agreement_ssfa(self):
+        self.agreement.agreement_type = Agreement.SSFA
+        self.agreement.save()
+        # self.agreement.
```
Comment (domdinicola) [inline src/etools/applications/partners/tests/test_views.py]: pls remove comment

File: src/etools/applications/partners/views/interventions_v2.py (hunk 1 above)
Comment (csabadenes) [reply src/etools/applications/partners/views/interventions_v2.py]: Done

File: src/etools/applications/partners/tests/test_views.py (hunk 2 above)
Comment (csabadenes) [reply src/etools/applications/partners/tests/test_views.py]: Done

**Pull request merged**